src/fabric/src/fabric2_db.erl

```diff
@@ -440,26 +455,32 @@ is_users_db(DbName) when is_binary(DbName) ->
     IsAuthCache orelse IsCfgUsersDb orelse IsGlobalUsersDb.
 
 
-set_revs_limit(#{} = Db, RevsLimit) ->
-    check_is_admin(Db),
+set_revs_limit(#{} = Db0, RevsLimit) ->
+    Db1 = require_admin_check(Db0),
     RevsLimBin = ?uint2bin(max(1, RevsLimit)),
-    Resp = fabric2_fdb:transactional(Db, fun(TxDb) ->
+    Resp = fabric2_fdb:transactional(Db1, fun(TxDb) ->
         fabric2_fdb:set_config(TxDb, <<"revs_limit">>, RevsLimBin)
     end),
-    if Resp /= ok -> Resp; true ->
-        fabric2_server:store(Db#{revs_limit := RevsLimit})
+    case Resp of
+        {ok, #{} = Db2} ->
+             fabric2_server:store(Db2#{revs_limit := RevsLimit});
```

Comment:
Sees awkward to return a `Db` variable from `set_config` without that config setting actually having been set. I know its set in the db, but seems like we should update the db there? Although that means we'd have to translate between binaries in atoms in set_config which seems also not awesome.

Reply:
Good point. It might look better if we return an already updated Db handle that has value already set.

Reply:
Updated with a separate commit which make set_config take already un-encoded values and returns a handle with an updated db version and the config value set.